packages/server-wallet/src/wallet/store.ts

```diff
@@ -260,6 +246,73 @@ export class Store {
     )?.protocolState;
   }
 
+  // Port of the following solidity code
+  // https://github.com/statechannels/statechannels/blob/a3d21827e340c0cc086f1abad7685345885bf245/packages/nitro-protocol/contracts/ForceMove.sol#L492-L534
+  async validateTransition(
+    fromState: SignedState,
+    toState: SignedState,
+    tx: Transaction
+  ): Promise<boolean> {
+    const fromMoverIndex = fromState.turnNum % fromState.participants.length;
+    const fromMover = fromState.participants[fromMoverIndex].signingAddress;
+
+    const toMoverIndex = toState.turnNum % toState.participants.length;
+    const toMover = toState.participants[toMoverIndex].signingAddress;
+    // This is the basic validation from the force move contract
+    // that applies to all state transitions
+    const basicValidation =
+      toState.turnNum === fromState.turnNum + 1 &&
+      toState.chainId === fromState.chainId &&
+      toState.participants === fromState.participants &&
+      toState.appDefinition === fromState.appDefinition &&
+      toState.challengeDuration === fromState.challengeDuration &&
+      fromState.signatures.some(s => s.signer === fromMover) &&
+      toState.signatures.some(s => s.signer === toMover);
```

Comment:
It's non-trivial to verify that this is a port of the solidity code. In fact, I don't think it is -- for example, this isn't exactly the signature check performed in the smart contracts.

Reply:
I modeled this based on the pseudocode @geoknee  has in the documentation. I chose that because it seemed clearer  than trying to mimic the contract which has multiple checks spread across multiple contracts. That does introduce the risk of a discrepancy but the rules seem simple enough 🤞